trio/_util.py

```diff
@@ -122,7 +122,7 @@ def __init__(self, func, args, kwds):
                 # Likewise, avoid suppressing if a StopIteration exception
                 # was passed to throw() and later wrapped into a RuntimeError
                 # (see PEP 479).
-                if exc.__cause__ is value:
+                if type is StopAsyncIteration and exc.__cause__ is value:
```

Comment:
I copied the code from the 3.7 version you linked and used "isinstance(value, (StopIteration, StopAsyncIteration)" which appeared to be equivalent to your suggestion.